R/utils.R

```diff
@@ -1161,3 +1144,122 @@ dashLogger <- function(event = NULL,
 clientsideFunction <- function(namespace, function_name) {
   return(list(namespace=namespace, function_name=function_name))
 }
+
+buildFingerprint <- function(path, version, hash_value) {
+  path <- file.path(path)
+  filename <- getFileSansExt(path)
+  extension <- getFileExt(path)
+  
+  sprintf("%s.v%sm%s.%s", 
+          file.path(dirname(path), filename),
+          gsub("[^\\w-]", "_", version, perl = TRUE),
+          hash_value,
+          extension)
+}
+
+checkFingerprint <- function(path) {
```

Comment:
Nonblocking. I was a bit confused about the fingerprint check for a few minutes because I did not see anything to handle `/` paths like in the Python impl, but it's being handled in the caller and only the filename portion of the path is passed here. Maybe renaming `path` to `filename` for clarity.

Reply:
Actually, the path is passed here -- hence the subsequent call to `basename`, which extracts the filename from the complete path if provided. I think I'll leave `path` as-is, though the earlier call to `checkFingerprint` could be written with `path` instead of `filename` to avoid confusion.